lib/dialects/abstract/query-generator.js

```diff
@@ -126,8 +126,9 @@ class QueryGenerator {
     if (this._dialect.supports['DEFAULT VALUES']) {
       emptyQuery += ' DEFAULT VALUES';
     } else if (this._dialect.supports['VALUES ()']) {
-      if (this.autoGenValue) {
-        emptyQuery += ` VALUES (${ this.autoGenValue++ })`;
+      // Db2 does not support VALUES () syntax. Add one value if not passed.
+      if (this.dialect === 'db2') {
+        emptyQuery += ` VALUES (${ options.uniqno++ })`;
```

Comment:
Hi, can you also explain please what is this `uniqno`? Also if Db2 does not support the `VALUES ()` syntax, why is it being considered inside an `if` statement that precisely cheks for `this._dialect.supports['VALUES ()']`? Thanks

Reply:
@papb Removed the code change in latest push.